src/librustc/mir/mod.rs

```diff
@@ -1149,11 +1149,13 @@ impl<'tcx> TerminatorKind<'tcx> {
             Return | Resume | Abort | Unreachable | GeneratorDrop => vec![],
             Goto { .. } => vec!["".into()],
             SwitchInt { ref values, switch_ty, .. } => {
+                let size = ty::tls::with(|tcx| switch_ty.scalar_size(tcx));
+                let size = size.map_or(0, |size| size.bits()) as u8;
                 values.iter()
                       .map(|&u| {
                           let mut s = String::new();
                           print_miri_value(
-                              Value::ByVal(PrimVal::Bytes(u)),
+                              Value::Scalar(Scalar::Bits { bits: u, defined: size }),
```

Comment:
Since this is for printing, could you pass in 128? There are certainly 128 defined bits, trivially.

Reply:
Oh this is used for sign extension o_O.

Reply:
I will still be passing this as a `Scalar`, because the function is also called elsewhere with other variants, but I fixed the sign extension thing